packages/react-select/src/Select.js

```diff
@@ -316,22 +482,20 @@ export default class Select extends Component<Props, State> {
     focusedValue: null,
     inputIsHidden: false,
     isFocused: false,
-    menuOptions: { render: [], focusable: [] },
     selectValue: [],
+    clearFocusValueOnUpdate: false,
+    inputIsHiddenAfterUpdate: undefined,
+    prevProps: undefined,
   };
 
   // Misc. Instance Properties
   // ------------------------------
 
   blockOptionHover: boolean = false;
   isComposing: boolean = false;
-  clearFocusValueOnUpdate: boolean = false;
   commonProps: any; // TODO
-  components: SelectComponents;
-  hasGroups: boolean = false;
```

Comment:
I removed this field because it was not being used any more (it's last usage was removed [here](https://github.com/JedWatson/react-select/pull/2295/files#diff-7064e1eb0f7889ae735e705d019ebfa160fbb668d7733f73ee6f107b8cf16fadL785)) and it wasn't trivial to figure out how to preserve its functionality with some of the changes I made.